backend/src/main/java/com/flightsearch/backend/Services/AccessTokenHandler.java

```diff
@@ -0,0 +1,24 @@
+package com.flightsearch.backend.Services;
+
+import java.io.IOException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.RequestScope;
+
+@Component
+@RequestScope
+public class AccessTokenHandler {
+
+  @Autowired
+  AccessTokenService accessTokenService;
+
+  private char[] tokenArray;
+
+  public String getAccessToken() throws IOException {
+    if (tokenArray == null) {
+      String accessToken = accessTokenService.getAccessToken();
+      tokenArray = accessToken.toCharArray();
```

Comment:
There are better and safer ways to cache a Bearer token, but as an example this was a good approach to improve the performance of the application.
For passwords (in this case token) it is good practice to keep the value as a char array instead of a String object.

You can check this [link](https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords#answer-8889285) for more information